### PR TITLE
Coap patches

### PR DESCRIPTION
**Pull request by KKopyscinski**
Fixes for coap testplan

### CODE REVIEW

**Comment by KKopyscinski** [comment]
@michal-narajowski could you review this?

File: defensics/coap_automation_handler.py [hunk 1]
```diff
@@ -50,8 +50,9 @@ def __init__(self):
             self.max_btmon_tries = 10
         self.result_path = None
         self.test_path = None
-        self.serial_out = None
+        self.rtt2pty = RTT2PTY()
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:55]: You create RTT2PTY object in 3 places. Is this necessary?

File: defensics/coap_automation_handler.py [hunk 2]
```diff
@@ -71,23 +72,27 @@ def process(self, job):
...
+                # make newtmgr instance and add connection
+                self.newtmgr = NewtMgr(profile_name='test', conn_type='oic_ble', connstring='"peer_name=c5"')
+                self.newtmgr.make_profile()
             logging.debug("Release lock")
             # consume buffered serial data by reading it and saving to /dev/null
             self.serial_out = SerialOutput(path=devnull,
                                            newtmgr=self.newtmgr)
-            self.serial_out.start()
-            self.serial_out.shutdown = True
+            self.serial_out.rtt2pty = RTT2PTY()
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:85]: It looks like the Serial object is now only used to hold rtt2pty object which seems unnecessary.

File: defensics/coap_proxy.py [hunk 3]
```diff
@@ -236,10 +236,10 @@ def send(self, data):
 
         future = self._ev_handler.wait_for_event(PROPERTIES_CHANGED,
                                                  self._notification_received)
-        try:
-            self.req_char_iface.WriteValue(data, {})
-        except dbus.DBusException:
-            pass
+        # try:
```
Comment (michal-narajowski) [inline defensics/coap_proxy.py:239]: This looks like WIP.

**Comment by KKopyscinski** [comment]
@michal-narajowski addressed the requested changes

File: defensics/coap_automation_handler.py [hunk 4]
```diff
@@ -71,25 +73,29 @@ def process(self, job):
...
+                self.newtmgr.make_profile()
             logging.debug("Release lock")
             # consume buffered serial data by reading it and saving to /dev/null
-            self.serial_out = SerialOutput(path=devnull,
-                                           newtmgr=self.newtmgr)
-            self.serial_out.start()
-            self.serial_out.shutdown = True
+            if serial_read_enable:
+                self.rtt2pty = RTT2PTY()
+            self.rtt2pty.rtt2pty_start()
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:84]: error: "None" has no attribute "rtt2pty_start" :)

File: defensics/coap_automation_handler.py [hunk 5]
```diff
@@ -157,6 +150,24 @@ def process(self, job):
             self.processing_lock.acquire()
             logging.debug("Acquire lock")
             logging.debug("Executing: after run")
+            self.rtt2pty.rtt2pty_stop()
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:153]: Ditto ^

**Review by michal-narajowski — CHANGES_REQUESTED** [review]

File: defensics/coap_automation_handler.py (hunk 4 above)
Comment (KKopyscinski) [reply defensics/coap_automation_handler.py]: now it shouldn't

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**